### PR TITLE
CMake reference

### PR DESCRIPTION
**Pull request by lasote**
Complete with https://github.com/conan-io/conan/pull/11178

### CODE REVIEW

File: examples/tools/cmake/cmake_toolchain/local_flow.rst [hunk 1]
```diff
@@ -0,0 +1,83 @@
+CMakeToolchain in the developer flow
+====================================
+
+.. note::
+
+    You can find the following example in this repository
+
+    .. code-block:: bash
+
+        $ git clone https://github.com/conan-io/examples2.git
```
Comment (czoido) [inline examples/tools/cmake/cmake_toolchain/local_flow.rst:10]: I'm not convinced about the idea that we make people clone a repository and cd to an empty folder, maybe we can skip this step for these cases but leave the test in the examples2 repository. Or just put the results of the `conan new` there and explaining in the example that it's the product of running `conan new...`

File: reference/tools/cmake/cmaketoolchain.rst [hunk 2]
```diff
@@ -1,10 +1,385 @@
...
+- **tools.cmake.cmaketoolchain:system_processor** is not necessary in most cases and is only used to force-define ``CMAKE_SYSTEM_PROCESSOR``.
+- **tools.build:cxxflags** list of extra C++ flags that will be appended to ``CMAKE_CXX_FLAGS_INIT``.
+- **tools.build:cflags** list of extra of pure C flags that will be appended to ``CMAKE_C_FLAGS_INIT``.
+- **tools.build:sharedlinkflags** list of extra linker flags that will be appended to ``CMAKE_SHARED_LINKER_FLAGS_INIT``.
+- **tools.build:exelinkflags** list of extra linker flags that will be appended to ``CMAKE_EXE_LINKER_FLAGS_INIT``.
+- **tools.build:defines** list of preprocessor definitions that will be used by ``add_definitions()``.
+- **tools.build:tools.apple:enable_bitcode** boolean value to enable/disable Bitcode Apple Clang flags, e.g., ``CMAKE_XCODE_ATTRIBUTE_ENABLE_BITCODE``.
+- **tools.build:tools.apple:enable_arc** boolean value to enable/disable ARC Apple Clang flags, e.g., ``CMAKE_XCODE_ATTRIBUTE_CLANG_ENABLE_OBJC_ARC``.
+- **tools.build:tools.apple:enable_visibility** boolean value to enable/disable Visibility Apple Clang flags, e.g., ``CMAKE_XCODE_ATTRIBUTE_GCC_SYMBOLS_PRIVATE_EXTERN``.
+
```
Comment (czoido) [inline reference/tools/cmake/cmaketoolchain.rst:385]: We should link the two examples in this page

File: examples/tools/cmake/cmake_toolchain/local_flow.rst [hunk 3]
```diff
@@ -0,0 +1,83 @@
+CMakeToolchain in the developer flow
```
Comment (czoido) [inline examples/tools/cmake/cmake_toolchain/local_flow.rst:1]: I personally don't see this so much as an example but as a part of the tutorial in the creating packages section where we explain with details what is the developer flow vs doing a `conan create` and

File: examples/tools/cmake/cmake_toolchain/local_flow.rst (hunk 1 above)
Comment (memsharded) [reply examples/tools/cmake/cmake_toolchain/local_flow.rst:10]: I agree, I think this is "core" enough to be in the main thread of the tutorial, not as a side example.

File: examples/tools/cmake/cmake_toolchain/local_flow_cmake_presets.rst [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+    The `build/generators` location is declared by the ``cmake_layout()`` in the ``layout()`` method.
+
+
+
+Building the project using ``CMakePresets``
+-------------------------------------------
+
+A ``CMakeUserPresets.json`` file is generated in the same folder of your ``CMakeLists.txt`` file,
+so you can use the ``--preset`` argument from ``cmake >= 3.23`` or use an IDE. The ``CMakeUserPresets.json`` is
```
Comment (czoido) [inline examples/tools/cmake/cmake_toolchain/local_flow_cmake_presets.rst:49]: I think this is worth adding a note on why >3.23

File: reference/tools/cmake/cmake_layout.rst [hunk 5]
```diff
@@ -1,7 +1,94 @@
...
+
+
+.. code:: python
+
+    from conan.tools.cmake import cmake_layout
+
+    def layout(self):
+        cmake_layout(self)
+
+If you want to try it, use the ``conan new hello/0.1 --template=cmake_lib`` template.
```
Comment (memsharded) [inline reference/tools/cmake/cmake_layout.rst:14]: Legacy
```suggestion
If you want to try it, use the ``conan new cmake_lib -d name=hello -d version=0.1`` template.
```

File: examples/tools/cmake/cmake_toolchain/local_flow_cmake_presets.rst [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+    You can find the following example in this repository
+
+    .. code-block:: bash
+
+        $ git clone https://github.com/conan-io/examples2.git
+        $ cd examples2/examples/tools/cmake/cmake_toolchain/local_flow_cmake_presets
+
+
+One of the advantages of using Conan toolchains is that they can help to achieve the same build
+with local development flows, as if the package is created in the cache.
```
Comment (czoido) [inline examples/tools/cmake/cmake_toolchain/local_flow_cmake_presets.rst:15]: I would add a more detailed introduction, explaining for example what's the advantage of using `CMakePresets, what IDE's support this, etc.

File: reference/tools/cmake/cmake.rst [hunk 7]
```diff
@@ -1,12 +1,65 @@
...
+
+        def build(self):
+            cmake = CMake(self)
+            cmake.configure()
+            cmake.build()
+
+
+.. note::
+
+    This helper includes the additional flag `-DCMAKE_SH="CMAKE_SH-NOTFOUND"` when using the `MinGW Makefiles` CMake's
```
Comment (memsharded) [inline reference/tools/cmake/cmake.rst:39]: Not anymore, this helper takes this CMAKE_SH variable from the CMakePresets.json file now

**Review by memsharded — COMMENTED** [review]
Great job.
I think I would move one of the examples to the main tutorial flow, but otherwise this is good stuff

File: reference/tools/cmake/cmaketoolchain.rst [hunk 8]
```diff
@@ -1,10 +1,385 @@
...
+            def build(self):
+                cmake = CMake(self)
+                cmake.configure()
+
+
+Extending and advanced customization
+------------------------------------
+
+Since Conan 1.36, ``CMakeToolchain`` implements a powerful capability for extending and customizing the resulting toolchain file.
+
```
Comment (czoido) [inline reference/tools/cmake/cmaketoolchain.rst:209]:
```suggestion
```

File: reference/tools/cmake/cmaketoolchain.rst [hunk 9]
```diff
@@ -1,10 +1,385 @@
...
+The following predefined blocks are available, and added in this order:
```
Comment (czoido) [inline reference/tools/cmake/cmaketoolchain.rst:210]: I think we should briefly explain what is a block?

**Review by memsharded — APPROVED** [review]

**Pull request merged**